src/category_theory/limits/shapes/images.lean

```diff
@@ -405,4 +417,109 @@ def im : arrow C ⥤ C :=
 
 end has_image_maps
 
+section strong_epi_mono_factorisation
+
+/-- A strong epi-mono factorisation is a decomposition `f = e ≫ m` with `e` a strong epimorphism
+    and `m` a monomorphism. -/
+structure strong_epi_mono_factorisation {X Y : C} (f : X ⟶ Y) :=
+(I : C)
+(e : X ⟶ I)
+(m : I ⟶ Y)
+(fac : e ≫ m = f)
+[e_strong_epi : strong_epi e]
+[m_mono : mono m]
+
+attribute [instance] strong_epi_mono_factorisation.m_mono strong_epi_mono_factorisation.e_strong_epi
+
+/-- Satisfying the inhabited linter -/
+instance strong_epi_mono_factorisation_inhabited {X Y : C} (f : X ⟶ Y) [strong_epi f] :
+  inhabited (strong_epi_mono_factorisation f) :=
+⟨⟨Y, f, 𝟙 Y, by simp⟩⟩
+
+/-- A strong epi-mono factorisation is in particular a mono factorisation. -/
+def strong_epi_mono_factorisation.to_mono_factorisation {X Y : C} {f : X ⟶ Y}
+  (F : strong_epi_mono_factorisation f) : mono_factorisation f :=
+⟨F.I, F.m, F.e, F.fac⟩
+
+/-- A mono factorisation coming from a strong epi-mono factorisation always has the universal
+    property of the image. -/
+def strong_epi_mono_factorisation.to_mono_is_image {X Y : C} {f : X ⟶ Y}
+  (F : strong_epi_mono_factorisation f) : is_image F.to_mono_factorisation :=
+{ lift := λ G, arrow.lift $ arrow.hom_mk' $ show G.e ≫ G.m = F.e ≫ F.m, by rw [F.fac, G.fac] }
+
+variable (C)
+
+/-- A category has strong epi-mono factorisations if every morphism admits a strong epi-mono
+    factorisation. -/
+class has_strong_epi_mono_factorisations :=
+(has_fact : Π {X Y : C} (f : X ⟶ Y), strong_epi_mono_factorisation.{v} f)
```

Comment:
I was slightly surprised seeing `fact` here, when we've otherwise abbreviated as `fac`.